frameworks/AutoGluon/exec.py

```diff
@@ -68,14 +69,21 @@ def run(dataset, config):
     # Persist model in memory that is going to be predicting to get correct inference latency
     predictor.persist_models('best')
 
+    def inference_time_classification(path: str):
+        data = TabularDataset(path)
+        return None, predictor.predict_proba(data, as_multiclass=True)
+
+    def inference_time_regression(path: str):
+        data = TabularDataset(path)
+        return predictor.predict(data, as_pandas=False), None
+
+    infer = inference_time_classification if is_classification else inference_time_regression
+    with Timer() as predict:
+        predictions, probabilities = infer(test_data)
     if is_classification:
-        with Timer() as predict:
-            probabilities = predictor.predict_proba(test_data, as_multiclass=True)
         predictions = probabilities.idxmax(axis=1).to_numpy()
-    else:
-        with Timer() as predict:
-            predictions = predictor.predict(test_data, as_pandas=False)
-        probabilities = None
+
+    inference_times = measure_inference_times(infer, dataset.inference_subsample_files)
```

Comment:
Unsure how realistic loading 1 row from a parquet file is for the real-time inference batch_size=1 scenario. I'd assume in most cases the data is part of a pipeline and thus is already expected to be in memory.

This is a minor point and I don't think it matters too deeply for the purposes of a paper.

Reply:
I don't think it's very realistic, likely the overhead of accessing the file dominates inference time for all but the most unwieldy models in this setting.  But for some frameworks it is the only way we can measure them (e.g., frameworks in other languages) unless we rely on self-reported numbers (if they provide any).

Reply:
Yeah, if you can add that for the Python frameworks, at least for batch size 1, that would be good. What we are really after is end-to-end latency for online inference scenarios where we can assume the data is already in memory (for example, being passed to a server endpoint with a warm-started model)

Reply:
I updated the PR with an example of how we could do this for AutoGluon. It's getting a bit unwieldy, but I think it works (read: will prioritise that the merge, refactor later).